src/providers/emoji/Emojis.cpp

```diff
@@ -296,31 +319,61 @@ std::vector<boost::variant<EmotePtr, QString>> Emojis::parse(
 
         for (const std::shared_ptr<EmojiData> &emoji : possibleEmojis)
         {
+            int emojiNonQualifiedExtraCharacters =
+                emoji->nonQualified.length() - 1;
             int emojiExtraCharacters = emoji->value.length() - 1;
-            if (emojiExtraCharacters > remainingCharacters)
+            if (remainingCharacters >= emojiExtraCharacters)
             {
-                // It cannot be this emoji, there's not enough space for it
-                continue;
-            }
+                // look in emoji->value
+                bool match = true;
 
-            bool match = true;
+                for (int j = 1; j < emoji->value.length(); ++j)
+                {
+                    if (text.at(i + j) != emoji->value.at(j))
+                    {
+                        match = false;
 
-            for (int j = 1; j < emoji->value.length(); ++j)
-            {
-                if (text.at(i + j) != emoji->value.at(j))
+                        break;
+                    }
+                }
+
+                if (match)
                 {
-                    match = false;
+                    matchedEmoji = emoji;
+                    matchedEmojiLength = emoji->value.length();
 
                     break;
                 }
             }
-
-            if (match)
+            else if (remainingCharacters >= emojiNonQualifiedExtraCharacters)
             {
-                matchedEmoji = emoji;
-                matchedEmojiLength = emoji->value.length();
+                // This checking here relies on the fact that the nonQualified string
+                // always starts with the same byte as value (the unified string)
+                bool match = true;
+
+                for (int j = 1; j < emoji->nonQualified.length(); ++j)
+                {
+                    if (text.at(i + j) != emoji->nonQualified.at(j))
+                    {
+                        match = false;
+
+                        break;
+                    }
+                }
+
+                if (match)
+                {
+                    matchedEmoji = emoji;
+                    matchedEmojiLength = emoji->nonQualified.length();
 
-                break;
+                    break;
+                }
+                // look in emoji->nonQualified
```

Comment:
This should probably be moved to the top of the block to match the block above.

Reply:
[`bc8e731` (#4840)](https://github.com/Chatterino/chatterino2/pull/4840/commits/bc8e7317a50217f1053dcd56d1008f6290909f27)